### DISCUSSION TITLE
Tabs with Lazy load

### DISCUSSION THREAD

**Discussion by mrahmadt** [filamentphp/filament#14240] [Help]
### Package

Infolist builder

### Package Version

3.2

### How can we help you?

I'm using Tabs very large data in each tab which make loading the page very slow (because filamentphp will render all tabs content when loading the page)

is there anyway to load Tab content only when the tab is active (Lazy load) and maybe destroy the content when the tab inactive?

Thanks

**Comment by leandrocfe** [comment]
You could achieve that using a custom Tab: https://filamentphp.com/docs/3.x/support/blade-components/tabs#overview

**Comment by mrahmadt** [comment]
Thanks @leandrocfe 

I saw the custom Tab, I didn't find anything mentioned about using lazy load.

I had to actually create custom page for each tab and use link in each tab to direct to the selected tab page

**Reply by leandrocfe** [reply]
I think this is the way to achieve that.
I think another solution would be creating a custom layout component extending the default `Tab` and customizing what you need
https://filamentphp.com/docs/3.x/forms/layout/custom

**Reply by LloydLabs** [reply]
How can we make this load lazily, though @leandrocfe ?

**Comment by binaryfire** [comment]
Hi @mrahmadt. Did you figure out a solution to this?

**Comment by RealMrHex** [comment]
Any solution on this guys on V4/V5?
@binaryfire @LloydLabs @leandrocfe @danharrin

**Reply by binaryfire** [reply]
I haven't found a clean way of doing this. I think it's kinda important though... One of the main uses for tabs in JS frameworks (at least for me) is lazy loading. Would love a `->lazy()` method on `Tab`.

Putting the lazy tab's content in its own Livewire component might work: https://www.answeroverflow.com/m/1316453680207233054. Haven't tried it though. And it's not so great for readability / maintenance.

**Reply by RealMrHex** [reply]
@binaryfire Thanks — I agree with you. This is an important feature for real-world systems.
We have many resources that use tabs in their forms and info lists, and each tab triggers multiple queries to load data. I’m really looking for a way to add lazy loading to reduce that overhead. The link you shared seems worth trying, especially since there’s no official solution for this yet.

**Reply by danharrin** [reply]
I would suggest using this, you can even make it look like tabs instead of a sidebar: https://filamentphp.com/docs/4.x/resources/overview#resource-sub-navigation

**Reply by RealMrHex** [reply]
Awesome, Thanks @danharrin 😎❤️

**Reply by howdu** [reply]
Another option could be a [skippable Wizard](https://filamentphp.com/docs/4.x/schemas/wizards#allowing-steps-to-be-skipped) to look like Tabs

**Reply by danharrin** [reply]
That still loads all steps at once

**Reply by indigoram89** [reply]
@danharrin thank you, but what if we already use sub navigation and we need to have something like lazy tabs in one of the sub navigation page? like sub-sub navigation =)  The case is Laravel Forge when we have Server resource with sub navigation on top of the content (Sites, Processes, ..., and Settings). So in the Settings tab (sub navigation page) we need also have tabs (on left side of the content) with different pages like SSH, Network, Redis. I think about it now a few hours but didn't find the solution.

**Reply by indigoram89** [reply]
This is an example what we need, I also recording a video course to learn Filament for my subscribers and we need a solution to do this case because example we doing need it.

![CleanShot 2025-12-03 at 20 03 50@2x](https://github.com/user-attachments/assets/bc15fa64-45a5-4e2a-8e7a-1f3b68ec01ea)
